File: RecorrerMatriz.java
import java.util.Scanner;

public class RecorrerMatriz {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int columnas = 3;
        int filas = 3;
        int[][] matriz = new int [filas][columnas];

        //capturar los numeros dentro del arreglo
        for (int i = 0; i < filas; i++) {
            for (int j = 0; j < columnas; j++) {
                System.out.println("ingrese un numero");
                matriz[i][j] = sc.nextInt();
            }
        }

        //mostrar los numeros dentro del arreglo
        for (int i = 0; i < filas; i++) {  
            for (int j = 0; j < columnas; j++) {
                System.out.println("el numero en la posicion [" + i + "][" + j + "] es: " + matriz[i][j]);
            }
        }
    }
}

File: RecorrerMatriz.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    int numeros[3][3];

    for (int i = 0; i <3; i++) {
        for (int j = 0; j <3; j++) {
            cout << "Ingrese un numero: ";
            cin >> numeros[i][j];
        }
    }
    cout << endl;
    
    for (int i = 0; i<3; i++ ) {
        for (int j = 0; j<3; j++) {
            cout << "el numero en la posicion [" << i << "] [" << j << "] es: " << numeros[i][j] <<endl;
        }
        cout << endl;
    }
}